### PR TITLE
Use TimedCache.Get() for read-only resources

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Use TimedCache.Get() for read-only resources
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Use TimedCache.Get() for read-only resources
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_vmssflex_cache.go [hunk 1]
```diff
@@ -322,7 +322,7 @@ func (fs *FlexScaleSet) DeleteCacheForNode(nodeName string) error {
 		return err
 	}
 
-	cached, err := fs.vmssFlexVMCache.Get(vmssFlexID, azcache.CacheReadTypeDefault)
+	cached, err := fs.vmssFlexVMCache.GetWithDeepCopy(vmssFlexID, azcache.CacheReadTypeDefault)
```
Comment (andyzhangx) [inline pkg/provider/azure_vmssflex_cache.go:325]: it will delete element in the map, so it should be non deep copy?

Comment (lzhecheng) [reply pkg/provider/azure_vmssflex_cache.go]: Updated.

File: pkg/provider/azure_vmss_cache.go [hunk 2]
```diff
@@ -280,15 +275,19 @@ func (ss *ScaleSet) DeleteCacheForNode(nodeName string) error {
 		return err
 	}
 
-	err = ss.vmssVMCache.Delete(getVMSSVMCacheKey(node.resourceGroup, node.vmssName))
+	// Delete VMSS cache
+	cached, err := ss.vmssCache.Get(consts.VMSSKey, azcache.CacheReadTypeUnsafe)
```
Comment (nilo19) [inline pkg/provider/azure_vmss_cache.go:279]: Can you rebase your branch? This is not aligned with the master.

Comment (lzhecheng) [reply pkg/provider/azure_vmss_cache.go:279]: Updated.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/retest

**Comment by lzhecheng** [comment]
`Lifecycle of VMSS should add node object when VMSS instance allocated` is quite unstable... Will investigate then.

**Comment by andyzhangx** [comment]
@lzhecheng could you rebase and make all tests pass?  thanks.

**Comment by lzhecheng** [comment]
> @lzhecheng could you rebase and make all tests pass? thanks.

Will do.

**Comment by andyzhangx** [comment]
close this PR since https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2949 has been merged

**Pull request closed without merging**